src/systemctl/systemctl-edit.c

```diff
@@ -309,6 +309,49 @@ static int unit_file_create_copy(
         return 0;
 }
 
+static void exec_editor(const char *editor, char **paths, size_t argc, char ***args_save) {
+        char **editor_args, **args, **original_path, **tmp_path;
+        size_t n_editor_args, i = 1;
+
+        args = *args_save;
+
+        if (isempty(editor))
+                return;
+
+        editor_args = strv_split(editor, WHITESPACE);
+        if (!editor_args) {
+                (void) log_oom();
+                _exit(EXIT_FAILURE);
+        }
+        n_editor_args = strv_length(editor_args);
+        if (n_editor_args == 0)
+                return;
+
+        argc += n_editor_args - 1;
+        args = reallocarray(args, argc + 1, sizeof(*args));
```

Comment:
this is a bit weird in order. usually when realloc()/reallocarray() is used, the result is assigned to some temporary var, and then when it is non-NULL it is used to override the original var. here you do it differently: you keep a shadow variable, that you immediately update, and then you sync it back to the original one, later

Reply:
but, what i don't get, afaics args is always NULL here on input, hence why use reallocarray() at all? a plain new() would suffice too. This also then makes the "args_save" argument to this function unnecessary